bin/main

```diff
@@ -1,98 +1,118 @@
-
-players[player2] = players[player1] == 'x' ? '0' : 'x'
-
-puts "#{player2} You have been assigned with #{players[player2]} symbol"
-p "Your wepon #{players}"
-
-puts "Here is the play board \n Rows are 1, 2 , 3 And Colums are a, b, c (mock)"
-puts '  _a___b__c_'
-puts '1 |  |  |  |'
-puts '  |__|__|__|'
-puts '2 |  |  |  |'
-puts '  |__|__|__|'
-puts '3 |  |  |  |'
-puts '  |__|__|__|'
-
-# Display the available space and let the player1 put x or 0
-
-def display__board
-  p @board[0]
-  p @board[1]
-  p @board[2]
+require_relative '../lib/board.rb'
+require_relative '../lib/player.rb'
+board = Board.new
+board.create_board
+puts "Welcome to tic tac toe
+1.play
+2.exit"
+choice = gets.chomp.to_i
+if choice == 2
+  at_exit { puts 'Goodbye, thanks for playing' }
+  exit
 end
-
-# This method search for position in board and change it for player symbol (TSE requirement)
-def position(number, symbol)
-  if number < 4
-    line = @board[0]
-  elsif number > 3 && number < 7
-    line = @board[1]
-  elsif number > 6
-    line = @board[2]
+validate = false
+puts 'What is you name Player 1?'
+name = gets.chomp
+
+puts "#{name} type a symbol to play with (X, O)"
+while validate == false
+  symbol = gets.chomp.upcase
+  if board.validate_symbol(symbol)
+    validate = true
   else
-    puts 'does not exist you lost your turn'
-
+    puts 'please enter a valid symbol'
   end
-  number = number.to_s
-  line&.each_with_index do |element, index|
-    line[index] = symbol if element == number
+end
+
+player1 = Player.new(name, symbol)
+
+puts 'What is your name Player 2?'
+name = gets.chomp
+symbol = symbol == 'X' ? 'O' : 'X'
+
+player2 = Player.new(name, symbol)
+
+puts "#{player2.name} You have been assigned with #{player2.symbol} symbol"
+
+def input(board)
+  validate = false
+
+  while validate == false
+    turn = gets.chomp.to_i
+    validate = board.validate_turn(turn)
+    puts 'please enter a valid position' if validate == false
   end
+  turn
 end
 
-turns = 0
-while turns < 9
-  # Now find out the available space and prompt player1 to put his symbol
-  puts "\n Its #{player1} turn now!"
-  puts "#{player1}, You can put #{players[player1]} in available spaces."
+# rubocop:disable Metrics/MethodLength
 
-  display__board
+def game(player1, player2, board)
+  turns = 0
+  while turns < 9
+    puts "\n Its #{player1.name} turn now!"
+    puts "#{player1.name}, You can put #{player1.symbol} in available spaces."
 
-  turn1 = gets.chomp.to_i
-  position(turn1, players[player1])
+    board.display_board
 
-  display__board
-  # Prompt player2 with available space to put his symbol
-  turns += 1
+    turn = input(board)
 
-  unless turns == 9
-    puts "\n Its #{player2} turn now!"
-    puts "#{player2}, You can put #{players[player2]} in available spaces."
+    board.place_symbol(turn, player1)
 
-    turn2 = gets.chomp.to_i
-    position(turn2, players[player2])
+    if board.check_win(player1.array)
+      puts "#{player1.name} won!"
```

Comment:
Let's take a closer look at your **game** method.
What does it do? It basically controls the main flow in our game.
How?
By cycling a logic between two players and applying the same rules to both of them.
:warning:  **Problems**

-  It is not DRY. Which means, it repeats itself. It does the same things for 2 players by using almost the same code. This logic can be extracted into another method and then you can call that method inside the **game** method, once for player1 and once for player2 and than iterate.
But this is not the only way to do it, I am giving just an example.

- It is not Single Responsible.

What does it do? 
1. It iterates, 
2. prints, 
3. taking action in case of a win situation

Every one of these responsibilities can be divided into isolated methods and then called inside it, to make your game method cleaner.

Reply:
### Thank you for the detailed analysis and the review. We got to know things in more clear way!

## Thanks to your constructive and detailed feedback! :bouquet: